shard.yml

```diff
@@ -5,11 +5,11 @@ authors:
   - Jesse Doyle <jdoyle@ualberta.ca>
 
 scripts:
-  postinstall: "make -C ext clean libduktape"
+  postinstall: make -C clean libduktape
```

Comment:
The `-C` flag changes the directory to the provided argument before running the provided targets.

As it currently stands, this may break the seamless `postinstall` for folk on non-windows platforms. Is there a way we can detect the environment from this shell command?

Reply:
Ah right, yeah that should be `make -C ext ...`.

Unfortunately there isn't, this is part of what makes Windows support difficult because Shards itself doesn't really have a concept of platform-specific script management. Some people have tried to work around this by having `postinstall` execute a Crystal file which handles platform-specific logic/configuration, but that may not be ideal here.

Reply:
I'm going to have to get a VM running Windows to test this - can we invoke a shell script here instead of `make` directly? If so, what type of shell does it run on in Windows?

i.e.

```yaml
postinstall: "./build/postinstall.sh"
```

Reply:
Windows' shell would be command prompt (`cmd`) but that can't run shell scripts. The (Git) Bash interpreter also does not ship with Windows by default, so there would be no way to execute the postinstall script on Windows.